src/internal/time.ts

```diff
@@ -0,0 +1,108 @@
+import { Fx, handle, ok } from '../Fx'
+import { Monotonic, Now, Schedule } from '../Time'
+
+export interface ScheduledTask {
+  readonly at: number
+  readonly task: () => void
+}
+
+export class TimeStep {
```

Comment:
I'm not crazy about the name `TimeStep`.  Suggestions welcome 😄

Reply:
The first thing that comes to mind to me is just `TestTime` just to connect `Time` and the idea that its for testing cases. Maybe `VirtualTime`, but I don't like that as much

Reply:
VirtualTime isn't bad.  TestTime is ok, too, although I'm not sure we should pigeon hole it as only for testing (even tho I can't imagine another use case atm).  Other ideas: ControlledTime, ExplicitTime

Reply:
or maybe use the word "Clock" here, like VirtualClock, TestClock, ControlledClock?

Reply:
It probably is more like a Clock than Time itself